docs/flow/concepts/components.md

```diff
@@ -6,3 +6,111 @@ weight: 100
 ---
 
 # Components
+
+_Components_ are the building blocks of Grafana Agent Flow. Each component is
+responsible for handling a single task, such as retrieving secrets or
+collecting Prometheus metrics.
+
+Components are composed of two parts:
+
+* Arguments: settings which configure a component
+* Exports: named values which a component exposes to other components
+
+Each component has a name which describes what that component is responsible
+for. For example, the `local.file` component is responsible for retrieving the
+contents of files on disk.
+
+Components are specified by users by first providing the component's name with
+a user-specified label, and then by providing arguments to configure the
+component.
```

Comment:
How about a sentence to explain that this is the way you can have multiple components of the same type?